# Descriptive Statistics

## Step 1: Loading the Dataset Column

The dataset was loaded using `pandas.read_csv()`. 
The `encoding="ISO-8859-1"` parameter was used to correctly read special characters in the CSV file and avoid encoding errors.

In [60]:
import numpy as np
import pandas as pd
df = pd.read_csv("Superstore.csv", encoding="ISO-8859-1")
df.head(5)



,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164



## Step 2: Computing Descriptive Statistics

The **Profit** column was selected because it represents the profit value for each order and can be analyzed using descriptive statistics.
Descriptive statistical measures were calculated for the Profit column, including:

- **Mean:** Represents the average profit value.
- **Median:** Represents the middle profit value after sorting the data.
- **Mode:** Represents the most frequent profit value.
- **Standard Deviation:** Measures how much the profit values vary from the mean.
- **IQR (Interquartile Range):** Measures the spread of the middle 50% of the data.


In [58]:
profit= df.Profit

mean =profit.mean()
median =profit.median()
mode =profit.mode()
std =profit.std(ddof=0)

print("Mean =", mean)
print("Median =", median)
print("Mode =", mode)
print("Standard Deviation =", std)

Mean = 28.65689630778467
Median = 8.6665
Mode = 0    0.0
Name: Profit, dtype: float64
Standard Deviation = 234.2483873603588


### Interquartile Range (IQR)

The **Interquartile Range (IQR)** measures the spread of the middle **50%** of the data.

**Formula:**
\[
IQR = Q3 - Q1
\]

Where:
- **Q1 (25th percentile)** = First Quartile
- **Q3 (75th percentile)** = Third Quartile

The IQR is commonly used to detect outliers using the following boundaries:

- **Lower Bound = Q1 - 1.5 × IQR**
- **Upper Bound = Q3 + 1.5 × IQR**

In [59]:
Q1=profit.quantile(.25)
Q2=profit.quantile(.5)
Q3=profit.quantile(.75)
IQR= Q3 - Q1
LB=Q1 - (1.5*IQR)
UB=Q3 + (1.5*IQR)
print("IQR =" , IQR)
print("Lower Bound =", LB)
print("Upper Bound =" , UB)




IQR = 27.63525
Lower Bound = -39.724125
Upper Bound = 70.816875


## Step 3: Choosing the Best Measure of Central Tendency

The mean profit value is approximately 28.66, while the median profit value is approximately 8.67.

Since the mean is much higher than the median and the standard deviation is large, this indicates that the Profit column contains extreme values (outliers) that affect the mean.

Therefore, the median is a better representation of the typical profit value because it is less affected by unusually high or low profit values.

## Step 4: Summary of Profit Distribution

The Profit column has an average value of approximately 28.66, with a median of approximately 8.67.

The large difference between the mean and median indicates that the data is not symmetrically distributed and contains some extreme profit values.

The standard deviation is approximately 234.25, showing that profits vary widely across different orders.

The IQR was calculated to measure the spread of the middle 50% of the data and was used to identify potential outliers.